goth_tests/README.md

```diff
@@ -0,0 +1,267 @@
+
+#### Docker engine
+To install Docker, follow [these instructions](https://docs.docker.com/engine/install/).
+To verify your installation you can run the following command:
+```
+docker run hello-world
+```
+
+#### Docker Compose
+`goth` tests require `docker-compose` **version 1.27** or higher. There are two ways you can install it:
+1. Download the appropriate executable from the [releases page](https://github.com/docker/compose/releases) and make sure its present on your system's `PATH`.
+2. Use the `docker-compose` installed in the virtual environment created by Poetry. You will need to activate the environment in the shell from which you run your tests. You can do so by running `poetry shell` from the `goth_tests` directory (you should see the env's name similar to `yagna-integration-tests--RlJJ9yy-py3.8` in your shell prompt).
+
+#### GitHub API token
+`goth` makes use of the GitHub API to download releases and artifacts for its test runs. Although all of these assets are public, using the GitHub API still requires basic authentication. Therefore you need to provide `goth` with a personal access token.
+
+To generate a new token, go to your account's [developer settings](https://github.com/settings/tokens).
+You will need to grant your new token the `public_repo` scope, as well as the `read:packages` scope. The packages scope is required in order to pull Docker images from GitHub.
+
+Once your token is generated you need to do two things:
+1. Log in to GitHub's Docker registry by calling: `docker login docker.pkg.github.com -u {username}`, replacing `{username}` with your GitHub username and pasting in your access token as the password. You only need to do this once on your development machine.
+2. Export an environment variable named `GITHUB_API_TOKEN` and use the access token as its value. This environment variable will need to be available in the shell from which you run the integration tests.
+
+### Running a test session
+With the project dependencies installed you are now ready to run some tests! All of the commands below assume you are running from the tests' root directory (`goth_tests`) and that you have a GitHub token available in your shell.
+
+First off, if you haven't done this already, you will need to generate `goth`'s default assets:
+```
+poetry run poe goth-assets
+```
+This will create a directory called `assets` in the tests' root directory. It contains a number of files which are either used by the `goth` runner or re-used by `yagna`'s test cases.
+
+These assets are ignored by `git` as their contents may change in the future (or be removed entirely) depending on `goth`'s version.
+
+With the assets generated, you can run the whole test suite by calling:
+```
+poetry run poe goth-tests
+```
+
+To run a single test you can call `pytest` directly like so:
+```
+pytest -svx e2e/vm/test_e2e_vm.py
```

Comment:
Better to compare `poetry run poe goth-tests` with `poetry run pytest -svx e2/vm/test_e2e_vm.py` here (or `poe goth-tests` with `pytest -svx e2e/vm/test_e2e_vm.py`, assuming the appropriate virtual env is activated).

Reply:
Changed this in 31cfad0